### ISSUE TITLE
Compatibility with OpenSesame 3.0

### ISSUE THREAD

**Issue by smathot** [dev-jam/opensesame_plugin_-_parallel_port_trigger#2]
I receive quite a few automated bug reports about this plug-in on OpenSesame 3.0. On the one hand this is good news, because it means people are using the plug-in. On the other, it means that it would be a good idea to update the plug-in. I suspect it's only a matter of fixing the management of experimental variables, so that they are all properties of the `var` object, as described here:

- http://osdoc.cogsci.nl/plug-ins/create-plug-in

**Comment by esdalmaijer** [comment]
Is `self.var` the new `self.experiment`?

**Comment by smathot** [comment]
Not really. `var` is a container object for experimental variables. Before, experimental variables where properties of the `experiment` object, but this made the distinction between experimental variables and the inner mechanics of OpenSesame very fuzzy.

**Comment by esdalmaijer** [comment]
Thanks! I've just seen @dev-jam's new implementation, and think I get the concept. Should probably change that in some of my old plug-ins too. Did you get any automatic bug reports on the slider plug-in? (Or anything else? I doubt people are still using my other things.)

**Comment by smathot** [comment]
Yes, I actually have seen a few bug reports for the slider as well.

**Comment by dev-jam** [comment]
Solved! Plugin is now opensesame 3 compliant (all my plugins btw)

At the moment I am using:
 `self.var` for item specific elements/variables
`self.experiment` for experiment wide variables/instances
`self.variable_name` for plugin wide variables

In the prepare phase I link the item specific variables to plugin wide variables:
self.variable_name1 = self.var.variable_name1
Could you advice me on this? Is this a good habit or can I better use the self.var through the entire class?
